app/main.py

```diff
@@ -1,3 +1,48 @@
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: Any) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> Any:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+
+    def __iadd__(self, other: Any) -> Any:
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            return NotImplemented
+        return self
+
+    def __mul__(self, num: int) -> Any:
+        return Distance(self.km * num)
+
+    def __truediv__(self, other: int) -> Any:
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: int) -> bool:
+        return self.km < other
+
+    def __gt__(self, other: int) -> bool:
+        return self.km > other
+
+    def __eq__(self, other: int) -> bool:
+        return self.km == other
+
+    def __le__(self, other: int) -> bool:
+        return self.km <= other
```

Comment:
The `__le__` method should compare `Distance` objects rather than integers. Update the method to check if `other` is an instance of `Distance` and compare `self.km` with `other.km`.